python/qpid_dispatch_internal/tools/display.py

```diff
@@ -45,7 +45,14 @@ def Commas(value):
     sval = left
 
 def TimeLong(value):
-  return strftime("%c", gmtime(value / 1000000000))
+  day = value // (24 * 3600)
+  time = value % (24 * 3600)
+  hour = time // 3600
+  time %= 3600
+  minutes = time // 60
+  time %= 60
+  seconds = time
+  return "%03d:%02d:%02d:%02d" % (day, hour, minutes, seconds)
```

Comment:
I'm curious as to why this was done.  Does the strftime function no format the data properly?

Reply:
I wanted qdstat to display in days:hours:minutes:seconds (we have a 3 digit limit on the days). I assumed that the router (or a connection) will not be up for more than 999 days.

That being said, strftime does not directly provide what I want. It rolls the the day into the year after 365 days. Then I need to use the year to calculate the days which becomes difficult if a leap year is involved. I could do the following and display the output in years:days:hours:minutes:seconds: 

value=150000000
year_from_epoch = strftime("%Y", gmtime(value))
year=int(year_from_epoch) - 1970
print ("Output=%s:%s" % (str(year), strftime("%j:%X", gmtime(value))))

[gmurthy@localhost test]$ python strtime.py 
Output=4:276:02:40:00
[gmurthy@localhost test]$ 

With all this in mind, I decided to write a simple custom function myself.